packages/braintree-integration/src/braintree-fastlane/braintree-fastlane-payment-strategy.spec.ts

```diff
@@ -626,6 +677,290 @@ describe('BraintreeFastlanePaymentStrategy', () => {
+            );
+
+            jest.spyOn(braintreeFastlaneMock.profile, 'showCardSelector').mockImplementation(
+                () => ({
+                    selectionChanged: false,
+                    selectedCard: {},
+                }),
+            );
+
+            let onChangeCallback: () => Promise<CardInstrument | undefined> = () =>
+                Promise.resolve(undefined);
+            const onChangeImplementation = (
+                showPayPalCardSelector: () => Promise<CardInstrument | undefined>,
+            ) => {
+                onChangeCallback = showPayPalCardSelector;
+            };
+
+            await strategy.initialize({
+                methodId,
+                braintreefastlane: {
+                    onInit: jest.fn(),
+                    onChange: jest.fn(onChangeImplementation),
+                },
+            });
+
+            const result = await onChangeCallback();
+
+            expect(result).toBeUndefined();
+        });
+
+        it('calls mapPayPalToBcInstrument', async () => {
+            paymentMethod.initializationData.isFastlaneEnabled = true;
+            jest.spyOn(braintreeFastlaneUtils, 'mapPayPalToBcInstrument');
+            jest.spyOn(braintreeFastlaneUtils, 'getBraintreeFastlaneOrThrow').mockReturnValue(
+                braintreeFastlaneMock,
+            );
+            jest.spyOn(
+                paymentIntegrationService.getState(),
+                'getPaymentProviderCustomer',
+            ).mockReturnValue({
+                authenticationState: BraintreeFastlaneAuthenticationState.SUCCEEDED,
+                addresses: [bcAddressMock],
+                instruments: [bcCardMock],
+            });
+
+            jest.spyOn(braintreeFastlaneMock.profile, 'showCardSelector').mockImplementation(
+                () => ({
+                    selectionChanged: true,
+                    selectedCard: {
+                        id: 'nonce/token',
+                        paymentSource: {
+                            card: {
+                                brand: 'Visa',
+                                expiry: '2030-12',
+                                lastDigits: '1111',
+                                name: 'John Doe',
+                                billingAddress: {
+                                    firstName: 'John',
+                                    lastName: 'Doe',
+                                    company: 'BigCommerce',
+                                    streetAddress: 'addressLine1',
+                                    extendedAddress: 'addressLine2',
+                                    locality: 'addressCity',
+                                    region: 'addressState',
+                                    postalCode: '03004',
+                                    countryCodeAlpha2: 'US',
+                                },
+                            },
+                        },
+                    },
+                }),
+            );
```

Comment:
```suggestion
beforeEach(() => {
           ...
           
            jest.spyOn(braintreeFastlaneUtils, 'mapPayPalToBcInstrument');
            jest.spyOn(braintreeFastlaneUtils, 'getBraintreeFastlaneOrThrow').mockReturnValue(
                braintreeFastlaneMock,
            );
            jest.spyOn(
                paymentIntegrationService.getState(),
                'getPaymentProviderCustomer',
            ).mockReturnValue({
                authenticationState: BraintreeFastlaneAuthenticationState.SUCCEEDED,
                addresses: [bcAddressMock],
                instruments: [bcCardMock],
            });

            jest.spyOn(braintreeFastlaneMock.profile, 'showCardSelector').mockImplementation(
                () => ({
                    selectionChanged: true,
                    selectedCard: {
                        id: 'nonce/token',
                        paymentSource: {
                            card: {
                                brand: 'Visa',
                                expiry: '2030-12',
                                lastDigits: '1111',
                                name: 'John Doe',
                                billingAddress: {
                                    firstName: 'John',
                                    lastName: 'Doe',
                                    company: 'BigCommerce',
                                    streetAddress: 'addressLine1',
                                    extendedAddress: 'addressLine2',
                                    locality: 'addressCity',
                                    region: 'addressState',
                                    postalCode: '03004',
                                    countryCodeAlpha2: 'US',
                                },
                            },
                        },
                    },
                }),
            );
        )}
```

You can move this code to beforeEach callback [here](https://github.com/bigcommerce/checkout-sdk-js/pull/2418/files#diff-afe7d0dc536a1ccc40b7e38c2d6611462b30e07cbc853673a29a85bad2964665R682) to avoid code duplication in several places